volume/profiles/profiles.go

```diff
@@ -234,33 +209,33 @@ func (pp *pvcVolProProfile) VSMName() (string, error) {
 }
 
 // ControllerCount gets the number of controllers
-func (pp *pvcVolProProfile) ControllerCount() (int, error) {
-	// Extract the controller count from pvc
-	return v1.GetPVPControllerCountInt(pp.pvc.Labels)
+func (pp *defVolProProfile) ControllerCount() (int, error) {
+	// Extract the controller count
+	return v1.GetPVPControllerCountInt(pp.vol.Labels)
 }
 
 // ControllerImage gets the controller's image currently its docker image label.
-func (pp *pvcVolProProfile) ControllerImage() (string, bool, error) {
-	// Extract the controller image from pvc
-	cImg := v1.GetControllerImage(pp.pvc.Labels)
+func (pp *defVolProProfile) ControllerImage() (string, bool, error) {
+	// Extract the controller image
+	cImg := v1.GetControllerImage(pp.vol.Labels)
 
 	return cImg, true, nil
 }
 
 // ReplicaImage gets the replica's image currently its docker image label.
-func (pp *pvcVolProProfile) ReplicaImage() (string, error) {
-	// Extract the replica image from pvc
-	rImg := v1.GetPVPReplicaImage(pp.pvc.Labels)
+func (pp *defVolProProfile) ReplicaImage() (string, error) {
+	// Extract the replica image
+	rImg := v1.GetPVPReplicaImage(pp.vol.Labels)
```

Comment:
I think it would be wise to catch the error while getting `PVPReplicaImage` (github.com/openebs/maya@HEAD/-/blob/types/v1/util.go#L789) and utilize that here.